python/admin/api.py

```diff
@@ -50,10 +50,8 @@ def set_system_property():
     if (not request.json) or ("value" not in request.json):
         _abort_bad_request() # bad request
 
-    if request.json["value"] in ["shutdown", "reboot"]:
-        # Safe as input is limited
-        subprocess.Popen("sleep 1; " + request.json["value"], shell=True)
-        return jsonify(code=0, result="SUCCESS")
+    if not request.json["value"] in ["shutdown", "reboot", "unmountusb", "reset"]:
+        _abort_bad_request() # bad request
```

Comment:
The `sleep 1` was here, for memory, because it allowed us to get a success/fail result back from the API. Why the change?

Reply:
So apparently what that code was doing was calling shutdown/reboot directly instead of passing the command through to the ConnectBoxManage.sh script. That differs from the php admin script and I don't think that's what we want. At the moment these functions don't work correctly because of that.

Reply:
great. thanks.